lib/puppet/file_serving/http_metadata.rb

```diff
@@ -20,7 +20,21 @@ def initialize(http_response, path = '/dev/null')
       checksum = checksum.unpack("m0").first.unpack("H*").first
       @checksums[:md5] = "{md5}#{checksum}"
     end
+    # Add support for artifactory headers.
+    # This adds support for all three algorithms, sha256, sha1, and md5.
+    if checksum = http_response['X-Checksum-Md5']
```

Comment:
I haven't seen the `X-Checksum-MD5` header before - usually I've seen this done via ETag header, but I don't think the HTTP spec says that ETag *must* be MD5, hence the need for an X style extension here it looks like.

I think these are mutually exclusive though, correct?  We can only serve one... which defaults to MD5 / is configurable - see https://github.com/puppetlabs/puppet/blob/master/lib/puppet/defaults.rb#L867-L887

This code doesn't quite look right to me...